routing/pathfind_test.go

```diff
@@ -616,6 +616,124 @@ func TestKShortestPathFinding(t *testing.T) {
 	assertExpectedPath(t, paths[1], "roasbeef", "satoshi", "luoji")
 }
 
+func TestNewRoute(t *testing.T) {
+
+	var sourceKey [33]byte
+	sourceVertex := Vertex(sourceKey)
+
+	const (
+		startingHeight = 100
+		finalHopCLTV   = 1
+	)
+
+
+	createHop := func(feeRate lnwire.MilliSatoshi, 
+		capacity btcutil.Amount) (*ChannelHop) {
+
+		return &ChannelHop {
+			ChannelEdgePolicy: &channeldb.ChannelEdgePolicy {
+				Node: &channeldb.LightningNode{},
+				FeeProportionalMillionths: feeRate,
+			},
+			Capacity: capacity,
+		}
+	}
+
+	testCases := []struct { 
+		name 		    string
+		hops 	    	    []*ChannelHop
+		paymentAmount 	    lnwire.MilliSatoshi
+		expectedTotalAmount lnwire.MilliSatoshi
+		expectError	    bool
+		expectedErrorCode   errorCode
+	} {
+	{
+		// For a single hop payment, no fees are expected to be paid
+		name: "single hop", 
+		paymentAmount: 100000,
+		hops: []*ChannelHop { 
+			createHop(1000, 1000),
+		},
+		expectedTotalAmount: 100000,
+	}, {
+		// For a two hop payment, only the fee for the first hop
+		// needs to be paid
+		name: "two hop", 
+		paymentAmount: 100000,
+		hops: []*ChannelHop { 
+			createHop(1000, 1000), 
+			createHop(1000, 1000),
+		},
+		expectedTotalAmount: 100100,
+	}, {
+		// Insufficient capacity in first channel when fees are added
+		name: "two hop insufficient", 
+		paymentAmount: 100000,
+		hops: []*ChannelHop { 
+			createHop(1000, 100), 
+			createHop(1000, 1000),
+		},
+		expectError: true,
+		expectedErrorCode: ErrInsufficientCapacity,
+	}, {
+		// A three hop payment where the first and second hop
+		// will both charge 1 msat. The fee for the first hop
+		// is actually slightly higher than 1, because the amount 
+		// t0 fwd also includes the fee for the second hop. This
+		// gets rounded down to 1.
+		name: "three hop", 
+		paymentAmount: 100000,
+		hops: []*ChannelHop { 
+			createHop(10, 1000), 
+			createHop(10, 1000), 
+			createHop(10, 1000),
+		},
+		expectedTotalAmount: 100002,
```

Comment:
maybe add a per hop expected fees as well. Will also make it easier to see how the total is calculated :)

Reply:
Indeed, maybe better to assert the full result. But I'd rather postpone that until #1321 is merged. Then I can properly separate the newRoute impl+test from the findPath impl+test. Different responsibilties, separate files.

Reply:
I don't see why we would have to postpone that. Looks like that could be added to these testcases without any dependency on #1321.